driverService/src/test/java/com/modsen/driverservice/integration/CarControllerIntegrationTest.java

```diff
@@ -0,0 +1,542 @@
+import static com.modsen.driverservice.util.ViolationData.LIMIT_EXCEEDED;
+import static com.modsen.driverservice.util.ViolationData.LIMIT_INSUFFICIENT;
+import static com.modsen.driverservice.util.ViolationData.OFFSET_INSUFFICIENT;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+import java.util.List;
+import java.util.Map;
+
+@Testcontainers
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class CarControllerIntegrationTest {
+
+    @Container
+    public static PostgreSQLContainer<?> postgreSQLContainer = new PostgreSQLContainer<>("postgres:13.3");
+
+    @DynamicPropertySource
+    static void postgreSQLProperties(DynamicPropertyRegistry registry) {
+        registry.add("spring.datasource.url", postgreSQLContainer::getJdbcUrl);
+        registry.add("spring.datasource.password", postgreSQLContainer::getPassword);
+        registry.add("spring.datasource.username", postgreSQLContainer::getUsername);
+    }
+
+    @Autowired
+    private CarRepository carRepository;
+
+    @Autowired
+    private DriverRepository driverRepository;
+
+    @Autowired
+    private WebApplicationContext webApplicationContext;
+
+    @Autowired
+    private JdbcTemplate jdbcTemplate;
+
+    @Autowired
+    ObjectMapper objectMapper;
+
+    static {
+        postgreSQLContainer.start();
+    }
+
+    @BeforeEach
+    void setUp() {
+        RestAssuredMockMvc.mockMvc(MockMvcBuilders.webAppContextSetup(webApplicationContext).build());
+        carRepository.deleteAll();
+        driverRepository.deleteAll();
+        jdbcTemplate.execute("ALTER SEQUENCE driver_id_seq RESTART WITH 1");
+        jdbcTemplate.execute("ALTER SEQUENCE car_id_seq RESTART WITH 1");
+
+        driverRepository.save(getDriver());
+        carRepository.save(getCar());
+    }
+
+    @Test
+    void getPageCars_whenEmptyParams_thenReturns201() {
+        RestAssuredMockMvc
+                .given()
+                .when()
+                .get(URL_CAR)
+                .then()
+                .statusCode(HttpStatus.OK.value())
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                .body(PAGE_NUMBER, equalTo(OFFSET_VALUE))
+                .body(PAGE_SIZE, equalTo(LIMIT_VALUE));
+    }
+
+    @Test
+    void getPageCars_whenValidParams_thenReturns201AndResponseDto() {
+        String responseJson = RestAssuredMockMvc
+                .given()
+                .param(OFFSET, OFFSET_VALUE)
+                .param(LIMIT, LIMIT_VALUE)
+                .when()
+                .get(URL_CAR)
+                .then()
+                .statusCode(HttpStatus.OK.value())
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                .extract()
+                .asString();
```

Comment:
Можно получить `PageDto<CarResponseDto>`:
```suggestion
        MockMvcResponse response = RestAssuredMockMvc
                .given()
                .param(OFFSET, OFFSET_VALUE)
                .param(LIMIT, LIMIT_VALUE)
                .when()
                .get(URL_CAR)
                .then()
                .statusCode(HttpStatus.OK.value())
                .contentType(MediaType.APPLICATION_JSON_VALUE);
 
                PageDto<CarResponseDto> actual = response.as(new TypeRef<PageDto<CarResponseDto>>() {});
```